### PR TITLE
Show tabs in the results when using the 'o' vomnibar

### PR DESCRIPTION
**Pull request by M-frankied**
Related to issue #2459.

Addresses a workflow like the following:
- Lots of tabs with e.g. issue tracker tickets, or admin web apps for lots of sites.
- I want to work on ticket x or site y, but I don't want x or y open in multiple tabs
- press 't', search for tab, not found
- close vomnibar
- press 'O', search for ticket or site

Getting access to the tab (if any) or otherwise the history item, bookmark, or search result from one vomnibar interaction is smoother here.

Changed the tab completer to have a quick weight falloff for subsequent tab entries to avoid crowding out other completion sources. This should have no influence on the workings of 't'.

### CODE REVIEW

**Comment by quintrino** [comment]
If I'm reading this correctly and it means that I can access both my open tabs and history/bookmarks from a single bar I'm very excited.

**Comment by deepredsky** [comment]
This is exactly what i have been looking for

File: background_scripts/completion.coffee [hunk 1]
```diff
@@ -364,19 +364,22 @@ class TabCompleter
...
           type: "tab"
           url: tab.url
           title: tab.title
-          relevancyFunction: @computeRelevancy
           tabId: tab.id
           deDuplicate: false
+        suggestion.relevancy = @computeRelevancy suggestion
+        suggestion
+      .sort (a,b) -> b.relevancy - a.relevancy
+      suggestions.forEach( (sugg,i) -> sugg.relevancy /= ( i / 4 + 1 ) )
```
Comment (smblott-github) [inline background_scripts/completion.coffee]: This computation requires comment explaining what and why.

Separately, elsewhere we use `suggestion` (not `sugg`) for an individual suggestion.

File: background_scripts/completion.coffee [hunk 2]
```diff
@@ -364,19 +364,22 @@ class TabCompleter
...
       onComplete suggestions
 
   computeRelevancy: (suggestion) ->
     if suggestion.queryTerms.length
-      RankingUtils.wordRelevancy(suggestion.queryTerms, suggestion.url, suggestion.title)
+      RankingUtils.wordRelevancy(suggestion.queryTerms, suggestion.url, suggestion.title) * 8.0
```
Comment (smblott-github) [inline background_scripts/completion.coffee]: The `* 8.0` here requires a comment/explanation.  Is this just a heuristic?

**Comment by j-martin** [comment]
@smblott-github is there are any plan on merging this?

**Comment by philc** [comment]
I just noticed that Chrome added support for this at some point, so that provides some additional rationale for mirroring their direction.

image

I'll test this out. Note that #2878 was just merged and is related.

**Comment by philc** [comment]
Here's what the current UX looks like on this PR. The new tab favicons look OK to me. The favicons help the tab results stand out more, which doesn't seem particularly negative. CC @gdh1995, @smblott-github for any additional thoughts.

image

My main concern is whether the tabs often overwhelm the rankings. I'll use this for a few days to get a feel for it.

**Comment by gdh1995** [comment]
To make it look better, we may give a same margin-left to URLs of history/bookmark items.

And now I have another idea to move icons before titles - currently my Vimium C looks like this:
image

**Comment by Qquanwei** [comment]
I like this feature so much . @philc Can we release this feature as soon as possible ?

**Comment by j-martin** [comment]
@philc thanks for looking into this. Worst come to worst, I think it could be option to show the tabs or not.

**Comment by M-frankied** [comment]
thanks @philc, and everyone contributing to this thread

**Pull request merged**